datafusion/functions/src/string/to_hex.rs

```diff
@@ -131,25 +118,37 @@ impl ScalarUDFImpl for ToHexFunc {
         &self.signature
     }
 
-    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        Ok(match arg_types[0] {
-            Int8 | Int16 | Int32 | Int64 | UInt8 | UInt16 | UInt32 | UInt64 => Utf8,
-            _ => {
-                return plan_err!("The to_hex function can only accept integers.");
-            }
-        })
+    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Utf8)
     }
 
     fn invoke_with_args(&self, args: ScalarFunctionArgs) -> Result<ColumnarValue> {
         match args.args[0].data_type() {
-            Int64 => make_scalar_function(to_hex::<Int64Type>, vec![])(&args.args),
-            UInt64 => make_scalar_function(to_hex::<UInt64Type>, vec![])(&args.args),
-            Int32 => make_scalar_function(to_hex::<Int32Type>, vec![])(&args.args),
-            UInt32 => make_scalar_function(to_hex::<UInt32Type>, vec![])(&args.args),
-            Int16 => make_scalar_function(to_hex::<Int16Type>, vec![])(&args.args),
-            UInt16 => make_scalar_function(to_hex::<UInt16Type>, vec![])(&args.args),
-            Int8 => make_scalar_function(to_hex::<Int8Type>, vec![])(&args.args),
-            UInt8 => make_scalar_function(to_hex::<UInt8Type>, vec![])(&args.args),
+            DataType::Null => Ok(ColumnarValue::Scalar(ScalarValue::Utf8(None))),
```

Comment:
I prefer having the `DataType::` but this is just personal preference, can revert if is too verbose